reactor-netty-core/src/main/java/reactor/netty/tcp/TcpClientConfig.java

```diff
@@ -142,12 +142,16 @@ protected ChannelPipelineConfigurer defaultOnChannelInit() {
 	 */
 	static final boolean SSL_DEBUG = Boolean.parseBoolean(System.getProperty(ReactorNetty.SSL_CLIENT_DEBUG, "false"));
 
-	static final class MicrometerTcpClientMetricsRecorder extends MicrometerChannelMetricsRecorder {
+	public static class MicrometerTcpClientMetricsRecorder extends MicrometerChannelMetricsRecorder {
 
 		static final MicrometerTcpClientMetricsRecorder INSTANCE = new MicrometerTcpClientMetricsRecorder();
 
-		MicrometerTcpClientMetricsRecorder() {
-			super(reactor.netty.Metrics.TCP_CLIENT_PREFIX, "tcp");
+		public MicrometerTcpClientMetricsRecorder(String prefix) {
+			super(prefix, "tcp");
+		}
+
+		public MicrometerTcpClientMetricsRecorder() {
+			this(reactor.netty.Metrics.TCP_CLIENT_PREFIX);
 		}
```

Comment:
Is there a value of making this public? One can directly use `MicrometerChannelMetricsRecorder`.

Reply:
Yes. The reason is the same for why we created a class `MicrometerTcpClientMetricsRecorder`, that's for abstraction.
If someday the implementation of `MicrometerTcpClientMetricsRecorder` changes, users don't need to know it because they can extend `MicrometerTcpClientMetricsRecorder`.